### DISCUSSION TITLE
How to make nuxt build to generate files is .js extension

### DISCUSSION THREAD

**Discussion by baobingkun** [nuxt/nuxt#16651] [Questions]
![image](https://user-images.githubusercontent.com/20108422/193167240-fe476c0a-8933-4ece-820b-c847fbdd1f20.png)
How to make nuxt build to generate files is .js extension. Because my project uses nginx proxy, nginx does not recognize .mjs, resulting in mjs not being cached. How to make the files generated by the nuxt build are all .js extensions

**Comment by sevillaarvin** [comment]
Can you clarify your use case? Are you using Nuxt as a server with nginx as proxy, or are you using nuxt as a static site generator with nginx as web server?

If you're using **Nuxt as a server**, nginx proxy should not care about the internal files of nuxt.

If you're using **Nuxt as a static site generator** for nginx web server, you should use `ssr: false` in your nuxt config: https://v3.nuxtjs.org/guide/deploy/static-hosting#static-hosting

**Reply by TheAlexLichter** [reply]
No, `ssr: false` is **not the right option when prerendering sites**. Then you want to keep SSR enabled and use `nuxi generate`

**Reply by sevillaarvin** [reply]
Forgot about that, thanks!

**Comment by TheAlexLichter** [comment]
The screenshot you've shown display the server files, which should be fine as MJS. Public assets should be JS files by now (since https://github.com/nuxt/framework/pull/6505).

**Reply by baobingkun** [reply]
Thanks.I solved my problem in a different way by modifying the nginx configuration so that mjs can be cached.

**Reply by TheAlexLichter** [reply]
That is another option ☺️
